projects/plugins/social/src/js/components/social-notes-toggle/index.tsx

```diff
@@ -99,6 +101,10 @@ const SocialNotesToggle: React.FC< SocialNotesToggleProps > = ( { disabled } ) =
 			checked={ isEnabled }
 			onChange={ toggleStatus }
 		>
+			{ ! isEnabled && (
+				// If social notes is disabled, hide the admin menu item, to avoid reloading the page
+				<style>{ `#adminmenu #menu-posts-jetpack-social-note { display: none; }` }</style>
+			) }
```

Comment:
This allows us to simply hide the Social Notes in the admin menu instead of doing the full page reload.